Add docs about non-linux binaries (#11889)
ClickHouse/ClickHouse: docs/en/getting-started/install.md

For production environments, it’s recommended to use the latest `stable`-version. You can find its number on GitHub page https://github.com/ClickHouse/ClickHouse/tags with postfix `-stable`.

> ### From precompiled binaries for non-Linux OS {#from-binaries-non-linux}

Reviewer: ```suggestion
### From Precompiled Binaries for Non-Linux OS {#from-binaries-non-linux}
```

> ### From precompiled binaries for non-Linux OS {#from-binaries-non-linux}

Reviewer: let's move all this below docker?

> For non-Linux operating systems, ClickHouse builds are provided as a cross-compiled binary from the latest commit of the master branch.

Reviewer: ```suggestion
For non-Linux operating systems, ClickHouse builds are provided as a cross-compiled binary from the latest commit of the `master` branch (with a few hours delay).
```

> - macOS — `curl -O 'https://clickhouse-builds.s3.yandex.net/master/macos/clickhouse'`

Reviewer: * let's also add clickable links to download from browser
* we'd better proxy those links through CDN so people Asia/America don't have to download from Moscow

> - AArch64 — `curl -O 'https://clickhouse-builds.s3.yandex.net/master/aarch64/clickhouse'`

Reviewer: it's for Linux, isn't it?

- FreeBSD — `curl -O 'https://clickhouse-builds.s3.yandex.net/master/freebsd/clickhouse'`

> These builds are not recommended for use in production environment. Not all ClickHouse features are present in them.

Reviewer: ```suggestion
These builds are not recommended for use in production environments because they are less thoroughly tested, but you can do so on your own risk. They also have only a subset of ClickHouse features available.
```
